src/packages/v4/utils/distributions.ts

```diff
@@ -0,0 +1,260 @@
+ * @returns Amount @amount minus the JB fee
+ */
+export function deriveAmountAfterFee(amount: number) {
+  return amount - amount * JB_FEE
+}
+
+/**
+ * Derive payout amount before the fee has been applied.
+ * @param amount - An amount that has already had the fee applied
+ * @returns Amount @amount plus the JB fee
+ */
+export function deriveAmountBeforeFee(amount: number) {
+  return amount / (1 - JB_FEE)
+}
+
+/**
+ * Derive payout amount from its % of the distributionLimit. Apply fee if necessary
+ * @param split - Payout split
+ * @param distributionLimit - Distribution limit
+ * @returns Amount that payout will receive as a number.
+ */
+export function derivePayoutAmount({
+  payoutSplit,
+  distributionLimit,
+  dontApplyFee,
+}: {
+  payoutSplit: Split
+  distributionLimit: number | undefined
+  dontApplyFee?: boolean
+}) {
+  if (!distributionLimit) return 0
+  const amountBeforeFee =
+    (payoutSplit.percent.toFloat() / ONE_BILLION) * distributionLimit
+  if (isJuiceboxProjectSplit(payoutSplit) || dontApplyFee) return amountBeforeFee // projects dont have fee applied
+  return deriveAmountAfterFee(amountBeforeFee)
+}
+
+/**
+ * Gets amount from percent of a bigger amount
+ * @param percent {float} - value as a percentage.
+ * @param amount string (hexString)
+ * @returns {number} distribution amount
+ */
+export function amountFromPercent({
+  percent,
+  amount,
+}: {
+  percent: number
+  amount: string
+}) {
+  return (percent / 100) * parseFloat(amount)
+}
+
+/**
+ * Gets split percent from split amount and the distribution limit
+ * @param amount {float} - value as a percentage.
+ * @param distributionLimit number
+ * @returns {number} percent as an actual percentage of distribution limit (/100)
+ */
+export function getDistributionPercentFromAmount({
+  amount, // Distribution amount before fee
+  distributionLimit,
+}: {
+  amount: number
+  distributionLimit: number
+}) {
+  return Number(splitPortionFromFormattedPercent((amount / distributionLimit) * 100))
+}
+
+/**
+ * Calculates sum of all split percentages
+ * @param splits {Split[]} - list of splits
+ * @returns {number} sum of all split percentanges
+ */
+export function getTotalSplitsPercentage(splits: Split[]) {
+  return splits.reduce(
+    (acc, curr) => acc + curr.percent.formatPercentage(),
+    0,
+  )
+}
+
+/**
+ * Due to limitations of rounding errors, it's possible that adjustedSplitPercents causes
+ * the splits to sum to 99.99999999% or 100.0000001% (causes error) instead of 100%.
+ * This function does one final pass of the percents to ensure they sum to 100%.
+ * @param splits {Split[]} - list of current splits to possibly have their percents adjusted
+ * @returns {Split[]} splits with their percents adjusted
+ */
+export function ensureSplitsSumTo100Percent({
+  splits,
+}: {
+  splits: Split[]
+}): Split[] {
+  // Calculate the percent total of the splits
+  const currentTotal = splits.reduce((sum, split) => sum + split.percent.toFloat(), 0)
```

Comment:
Just a comment, no need to fix now: wouldn't it be more accurate to use bigint throughout this function?

Reply:
It would probably be more clear, but IIRC, the frontend code is as precise as it possibly can be in terms of rounding. It's actually limited by the contract payouts which support ~10dp. 

Even at this level of precision sometimes situations can arise where we try to submit payouts to the contracts where the total payout percent is 100.000000001, which causes a error.  So this function is a final safeguard against that.